docs/_tooling/decision-records/multirepo.md

```diff
@@ -0,0 +1,116 @@
+* In score versioning of requirement-links is handled via hashes.
+* We have two different mechanisms for versioning. Current assumption is that we'll use bazel to pull other repositories in a specific version, while we never pull different versions of the same repository. So basically, we have the "classic multi repo setup" situation.
+
+
+## Alternatives
+
+These are the identified alternatives. We'll dive into each one in detail.
+1) weblinks
+2) needservice
+3) needimport
+4) Needs-external-needs
+5) Copying
+
+### 1) needservice
+
+This is basically a manual approach to the problem. As long as any other solution works,
+that would be preferrable.
+
+### 2) weblinks
+
+We can simply link pages / needs in the other repositories by their full url.
+While we can ensure that those links work, everything beyond that will become problematic.
+Versioning might be solvable, but checking correct hashes (versioning) would be challenging.
+
+Bidirectional links are not possible. As we'd like the same approach everywhere, this is a **no-go**.
+
+### 3) needimport
+
+This is a sphinx-needs extension that allows to link to needs in other repositories.
+The other repositories do NOT need to be available at build time. Only their build output is required (needs.json).
+
+needs from the other repositories are imported, as if they were local.
+All structure is lost! All surrounding text, images etc are lost. Only the needs themselfes are imported. This is a **no-go**.
+
+
+### 4) Needs-external-needs (+ intersphinx)
+
+This is a sphinx-needs extension that allows to link to needs in other repositories.
+The other repositories do NOT need to be available at build time. Only their build output is required (needs.json).
+
+For bidirectional linking:
+* Build all repositories once, so we have needs.json
+* Exchange needs.json
+* Build all repositories again
+
+As is already evident in the requirements, we need multiple versions of the score repository website. One per integration. Since the pure score repository website shall not contain links to the other repositories, except when used in the context of an integration.
+
+Pro:
+* Bidirectional links are possible
+* Good performance (no need to build all repositories at once)
+Con:
+* needs.json and the entire website needs to be hosted in multiple versions.
+* bidirectional links affect the "core score" repository, which is not desired.
+
+Within the bounds of the requirements, this is a **no-go**.
+
+### 5) Copying
+
+We use bazel to "import" the other repositories.
+When building a module or even an integration, everything is build in one go, into one website.
+
+Pro:
+* Bidirectional links are possible
+* In reference implementation the versioning is handled by bazel for source code anyway. Since docs are located next to the source code, Version handling for docs is for free!
+
+Con:
+* Relies heavily on bazel -> potential problems with esbonio etc
+* Performance / runtime
```

Comment:
This problem can be huge, depending on the content of the docs.
For instance, if each module-repo needs a doxygen/doxysphinx beuild for API -documentation.

It may be okay for a CI build, but giving the user feedback during editing (esbonio build) would take too long.

Maybe a 2-way approach/support would be good:
* A "complete" build for the CI and the deployed documentation
* A local quick & dirty build for the specific repo only, in which the current change gets implemented (for quick user feedback of ~95% of our checks/rules.)

Reply:
Resolving, as the text is completely rewrtten.